### ISSUE TITLE
WinML or DirectML support planned?

### ISSUE THREAD

**Issue by oscarbg** [Microsoft/CNTK#3103]
Windows only, but this should support AMD and Intel GPUs using D3D12 backend  or even DirectML which with custom  metacommands seems even able to use tensor cores on Volta GPUs..
Hope to hear news at least by Build conference in May

**Comment by veikkoeeva** [comment]
Cross-referencing https://github.com/Microsoft/CNTK/issues/960 and https://github.com/Microsoft/CNTK/issues/2839 as relevant.

**Comment by oscarbg** [comment]
@veikkoeeva yes not really needed WinML is a Win32 API in Windows insider SDKs and this month 1803 SDK it has C++ header WinML.h and WinML.lib.. altough still no samples using Win32 API presumably at Build some Win32 API samples released..
See how Win32 API:
![847d82e6-609c-4f88-9c94-0b4985a17582](https://user-images.githubusercontent.com/669305/38461413-8054ecd6-3acf-11e8-8fbc-76396f9fcbf1.jpeg)
Taken from (with DirectML details):
https://blogs.msdn.microsoft.com/directx/2018/03/19/gaming-with-windows-ml/

**Comment by veikkoeeva** [comment]
It appears ONNX and WinML are in https://github.com/dotnet/machinelearning/.
